clients/python/templates/client.mustache

```diff
@@ -12,20 +12,35 @@ from {{apiPackage}} import {{classFilename}}
 class _WrappedApiClient(ApiClient):
     """ApiClient that fixes some weirdness"""
 
-    # Wrap files_parameters to work with unnamed "files" (e.g. MemIOs).
     def files_parameters(self, files=None):
-        if files is not None:
-            for (param_name, file_instances) in files.items():
-                i = 0
-                if file_instances is None:
-                    continue
-                for file_instance in file_instances:
-                    if file_instance is not None and not hasattr(file_instance, 'name'):
-                        # Generate a fake name.
-                        i += 1
-                        file_instance.name = f'{param_name}{i}'
-        return super().files_parameters(files)
+        """
+        Transforms input file data into a formatted list to return file_parameters.
+        Assume a string file_name is a path to the file to read.
+        Assume a bytes file_name is a file-like object that we append the information.
+        The parent class will handle the files to read.
+        """
+        if not files:
+            return []
 
+        params = []
+        files_to_read = {}
+
+        for idx, (key, value) in enumerate(files.items()):
+            if not value:
+                continue
+
+            # Ensure the value is always a list.
+            file_names = value if isinstance(value, list) else [value]
+
+            for file_name in file_names:
+                if type(file_name) is str:
+                    files_to_read[key] = file_name
+                else:
+                    name = f'{key}{idx}'
```

Comment:
Not sure this guarantees unique names (if that is the intent).  For that you need a nondigit separator for the suffix, e.g.
```suggestion
                    name = f'{key}_{idx}'
```

Reply:
I'll take this change, but it is per request and we currently use single file upload for upload_object (which the only api that uses files) - so I think we can't conflict with actual filename in the same call.